### PR TITLE
Fix to Cannot add same file after cancelled #21

### PR DESCRIPTION
**Pull request by maxwkf**

### CODE REVIEW

File: js/backbone.upload-manager.js [hunk 1]
```diff
@@ -465,7 +478,15 @@
             computeData: function ()
             {
                 return $.extend(this.getHelpers(), this.model.get('data'));
-            }
+            },
+            
+            // Max Wong: Cannot add same file after cancelled #21
+            _clearPreviousAddFileInput: function() {
```
Comment (sroze) [inline js/backbone.upload-manager.js]: Can you remove the copy/pasting of this method ?

File: js/backbone.upload-manager.js [hunk 2]
```diff
@@ -69,6 +69,13 @@
             // Add local events handlers
             this.bindLocal();
         },
+        // Max Wong: Cannot add same file after cancelled #21
```
Comment (sroze) [inline js/backbone.upload-manager.js]: Git blame will give you the authorship, no worries you can remove these comments :wink:

File: js/backbone.upload-manager.js [hunk 3]
```diff
@@ -69,6 +69,13 @@
...
+        _clearPreviousAddFileInput: function() {
+        	var self = this;
```
Comment (sroze) [inline js/backbone.upload-manager.js]: Why is that relevant to us a `self` variable here when you can directly use `this` ?

File: js/backbone.upload-manager.js [hunk 4]
```diff
@@ -450,11 +459,15 @@
                 // DOM events
                 $('#btn-cancel', this.el).click(function(){
                     self.model.cancel();
-                    self.collection.remove(self.model);
```
Comment (sroze) [inline js/backbone.upload-manager.js:453]: So you aren't removing anything from the collection ? \o/

File: js/backbone.upload-manager.js (hunk 2 above)
Comment (maxwkf) [inline js/backbone.upload-manager.js]: Haha, it is just my personal practice to add a name if a program is collaborate by multiple programmer.
This is more easy to trace who make this silly code later.

Just remove it when you combine to your version.

File: js/backbone.upload-manager.js (hunk 1 above)
Comment (maxwkf) [inline js/backbone.upload-manager.js]: Um... I think I am to green to refactor it.
I dont know how to make it across 2 objects.

File: js/backbone.upload-manager.js (hunk 3 above)
Comment (maxwkf) [inline js/backbone.upload-manager.js]: Yes you are right. I make a stupid code here....lol

File: js/backbone.upload-manager.js [hunk 5]
```diff
@@ -205,6 +205,7 @@
                 while (self.files.length) {
                     self.files.at(0).cancel();
                 }
+                self.$('#' + self.options.fileUploadId).val('');
```
Comment (maxwkf) [inline js/backbone.upload-manager.js:208]: Just removed the _clearPreviousAddFileInput function and directly set to empty value.

**Pull request closed without merging**